base/compiler/tfuncs.jl

```diff
@@ -885,8 +890,66 @@ function getfield_tfunc(@nospecialize(s00), @nospecialize(name))
     return rewrap_unionall(R, s00)
 end
 
-setfield!_tfunc(o, f, v, order) = (@nospecialize; v)
-setfield!_tfunc(o, f, v) = (@nospecialize; v)
+function setfield!_tfunc(o, f, v, order)
+    @nospecialize
+    if !isvarargtype(order)
```

Comment:
I thought we only ever call tfuncs with real types

Reply:
Unfortunately `argtypes[end]` can sometimes be varargtype:
https://github.com/JuliaLang/julia/blob/ee72a2bbddfdf4a5be4fd173201e7d5e0c64e0ab/base/compiler/tfuncs.jl#L1682

Reply:
Maybe we want to try to make sure that most tfuncs deal with real types?